src/ert/run_models/base_run_model.py

```diff
@@ -503,12 +503,14 @@ async def run_monitor(
                         EESnapshotUpdate,
                     }:
                         event = cast(EESnapshot | EESnapshotUpdate, event)
-                        await asyncio.get_running_loop().run_in_executor(
-                            None,
-                            self.send_snapshot_event,
-                            event,
-                            iteration,
-                        )
+                        if asyncio.get_running_loop().is_running():
```

Comment:
Hm, this I think is redundant as this should be always true as this runs inside `asyncio.run(...)`
I would recommend instead:
```python
loop = asyncio.get_running_loop()
if loop.is_closed():
   self.send_snapshot_event(event, iteration)
else:
   await asyncio.get_running_loop().run_in_executor(
    None,
    self.send_snapshot_event,
    event,
    iteration,
)
```